### PR TITLE
Starkware

### PR DESCRIPTION
**Pull request by NiklasKunkel**
This PR is functional but isn't meant to be merged as-is. Rather, it is to start a review and clean up the code. Additionally the following dependencies will need to be added to nix:

python3
python3-mpmath
python3-sympy.

We also will want to include 
ecdsa 0.16 

after which we can rename signature_ecdsa_0_16_0_dependent.py to signature.py and remove the current signature.py which is a much messier version of the code that does not depend on ecdsa.

### CODE REVIEW

File: omnia/scuttlebot.sh [hunk 1]
```diff
@@ -76,13 +79,20 @@ broadcastPriceMsg () {
...
+    #format starkware sig
+
+    _starkSignature=( "--arg r $_starkSignatureR" "--arg s $_starkSignatureS" "--arg publicKey $_starkPublicKey" )
+    verbose  "${_starkSignature[*]}"
+    if ! _starkSignatureJson=$(jq -nce  --arg r "$_starkSignatureR" --arg s "$_starkSignatureS" --arg publicKey "$_starkPublicKey" '{r: $r, s:$s, publicKey:$publicKey}'); then
+        error "Error - failed to generate stark signature json"
+    fi
     #compose jq message arguments
-    _jqArgs=( "--arg assetPair $_assetPair" "--arg version $OMNIA_VERSION" "--arg price $_price" "--arg priceHex $_priceHex" "--arg time $_time" "--arg timeHex $_timeHex" "--arg hash ${_hash:2}" "--arg signature ${_signature:2}" "--argjson sourcePrices $_sourcePrices" )
+    _jqArgs=( "--arg assetPair $_assetPair" "--arg version $OMNIA_VERSION" "--arg price $_price" "--arg priceHex $_priceHex" "--arg time $_time" "--arg timeHex $_timeHex" "--arg hash ${_hash:2}" "--arg signature ${_signature:2}" "--argjson starkSignature $_starkSignatureJson" "--argjson sourcePrices $_sourcePrices" )
```
Comment (icetan) [inline omnia/scuttlebot.sh:90]: Same as for `$_starkSignature`.

```suggestion
    _jqArgs=( --arg assetPair "$_assetPair" --arg version "$OMNIA_VERSION" --arg price "$_price" --arg priceHex "$_priceHex" --arg time "$_time" --arg timeHex "$_timeHex" --arg hash "${_hash:2}" --arg signature "${_signature:2}" --argjson starkSignature "$_starkSignatureJson" --argjson sourcePrices "$_sourcePrices" )
```

File: omnia/scuttlebot.sh [hunk 2]
```diff
@@ -76,13 +79,20 @@ broadcastPriceMsg () {
...
     #debug
     verbose "${_jqArgs[*]}"
     #generate JSON msg
     # shellcheck disable=2068
```
Comment (icetan) [inline omnia/scuttlebot.sh:94]: Remove this as it warns about splitting parameter substitutions which should be fixed, see next comment.

```suggestion
```

File: omnia/scuttlebot.sh [hunk 3]
```diff
@@ -76,13 +79,20 @@ broadcastPriceMsg () {
...
-    if ! _json=$(jq -ne ${_jqArgs[@]} '{type: $assetPair, version: $version, price: $price | tonumber, priceHex: $priceHex, time: $time | tonumber, timeHex: $timeHex, hash: $hash, signature: $signature, sources: $sourcePrices}'); then
+    if ! _json=$(jq -ne ${_jqArgs[@]} '{type: $assetPair, version: $version, price: $price | tonumber, priceHex: $priceHex, time: $time | tonumber, timeHex: $timeHex, hash: $hash, signature: $signature, starkSignature: $starkSignature, sources: $sourcePrices}'); then
```
Comment (icetan) [inline omnia/scuttlebot.sh:95]: Fix problem with splitting parameter substitutions, see Shellcheck warning: https://github.com/koalaman/shellcheck/wiki/SC2068

```suggestion
    if ! _json=$(jq -ne "${_jqArgs[@]}" '{type: $assetPair, version: $version, price: $price | tonumber, priceHex: $priceHex, time: $time | tonumber, timeHex: $timeHex, hash: $hash, signature: $signature, starkSignature: $starkSignature, sources: $sourcePrices}'); then
```

**Review by icetan — CHANGES_REQUESTED** [review]
Review done by me, @MDobak @pawzar and @konstantinzolotarev 

I think we can remove the `__pycache__` directory and add it to the `.gitignore`. These files are generated at runtime and should not be commited.

**Review by icetan — APPROVED** [review]
:star2:

**Comment by ViviG69** [comment]
Hello

**Pull request merged**